executor/src/native_executor.rs

```diff
@@ -0,0 +1,110 @@
+use std::panic::catch_unwind;
+use primitives::contract::CallData;
+use state_machine::{Externalities, CodeExecutor};
+use error::{Error, ErrorKind, Result};
+use wasm_executor::WasmExecutor;
+use native_runtime as runtime;
+use runtime_std;
+
+pub struct NativeExecutor;
+
+fn safe_call<F: ::std::panic::UnwindSafe + FnOnce() -> Vec<u8>>(f: F) -> Result<Vec<u8>> {
+	catch_unwind(f).map_err(|_| ErrorKind::Runtime.into())
+}
+
+impl CodeExecutor for NativeExecutor {
+	type Error = Error;
+
+	fn call<E: Externalities>(
+		&self,
+		ext: &mut E,
+		code: &[u8],
+		method: &str,
+		data: &CallData,
+	) -> Result<Vec<u8>> {
+		// WARNING!!! This assumes that the runtime was built *before* the main project. Until we
+		// get a proper build script, this must be strictly adhered to or things will go wrong.
+		let native_equivalent = include_bytes!("../../wasm-runtime/target/wasm32-unknown-unknown/release/runtime_polkadot.compact.wasm");
+		if code == &native_equivalent[..] {
+			runtime_std::with_externalities(ext, || match method {
+				"execute_block" => safe_call(|| runtime::execute_block(&data.0)),
+				"execute_transaction" => safe_call(|| runtime::execute_transaction(&data.0)),
```

Comment:
I am little worried about this since `&mut T` isn't `UnwindSafe`. But this code compiles because we've hidden the mutable reference in the `with_externalities` global.

Doesn't have to be for this PR, but the entire runtime should eventually change to not use panics unless the program should actually halt.